shell/platform/android/io/flutter/view/FlutterMain.java

```diff
@@ -271,22 +281,41 @@ public void run() {
      */
     private static void initConfig(Context applicationContext) {
         try {
-            Bundle metadata = applicationContext.getPackageManager().getApplicationInfo(
-                applicationContext.getPackageName(), PackageManager.GET_META_DATA).metaData;
+            Bundle metadata = applicationContext
+                .getPackageManager()
+                .getApplicationInfo(applicationContext
+                    .getPackageName(), PackageManager.GET_META_DATA)
+                .metaData;
+
             if (metadata != null) {
                 sAotSharedLibraryPath = metadata.getString(PUBLIC_AOT_AOT_SHARED_LIBRARY_PATH, DEFAULT_AOT_SHARED_LIBRARY_PATH);
-                sAotVmSnapshotData = metadata.getString(PUBLIC_AOT_VM_SNAPSHOT_DATA_KEY, DEFAULT_AOT_VM_SNAPSHOT_DATA);
-                sAotVmSnapshotInstr = metadata.getString(PUBLIC_AOT_VM_SNAPSHOT_INSTR_KEY, DEFAULT_AOT_VM_SNAPSHOT_INSTR);
-                sAotIsolateSnapshotData = metadata.getString(PUBLIC_AOT_ISOLATE_SNAPSHOT_DATA_KEY, DEFAULT_AOT_ISOLATE_SNAPSHOT_DATA);
-                sAotIsolateSnapshotInstr = metadata.getString(PUBLIC_AOT_ISOLATE_SNAPSHOT_INSTR_KEY, DEFAULT_AOT_ISOLATE_SNAPSHOT_INSTR);
                 sFlutterAssetsDir = metadata.getString(PUBLIC_FLUTTER_ASSETS_DIR_KEY, DEFAULT_FLUTTER_ASSETS_DIR);
+                aotAbiSplit = metadata.getBoolean(PUBLIC_AOT_ABI_SPLIT_KEY, DEFAULT_AOT_ABI_SPLIT);
```

Comment:
Does this require declaring use of AotAbiSplit mode in the application's manifest?

Would it be possible to check for the presence of the blob libraries and then fall back to blob assets if the libraries are not found?

Reply:
I think that seems safer as well - the manifest key could end up falling out of sync, and would require some kind of migration magic in the tooling or gradle plugin later on (e.g. I created my app using Flutter stable today, and when I get the next stable version with this logic I now have to manually add that key to get it to run).

Reply:
Definitely. I didn't think about that. Thanks for pointing it out!